File: backend/src/main/java/top/vuhe/drive/DataFrame.java
package top.vuhe.drive;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

/**
 * 信息帧
 * <p>
 * 从单片机设备获取的信息帧, 其中包含:
 * 命令数据: 包含 1byte, 是设备控制信息
 * dataInfo: 包含 [0, N]byte，是设备发送的数据
 *
 * @author zhuhe
 */
@Data
@AllArgsConstructor
class DataFrame {
    private int command;
    private List<Byte> dataInfo;
}

File: backend-kt/src/main/kotlin/top/vuhe/sw/drive/DataFrame.kt
package top.vuhe.sw.drive

/**
 * ## 信息帧
 *
 * 从单片机设备获取的信息帧, 其中包含:
 *
 * 命令数据: 包含 1byte, 是设备控制信息
 *
 * dataInfo: 包含 [0, N]byte，是设备发送的数据
 *
 * @property command  命令
 * @property dataInfo 字节信息
 */
data class DataFrame(
    val command: CommandEnum,
    val dataInfo: List<Byte>
)
